install

```diff
@@ -96,6 +101,11 @@ while true; do
     esac
     done
 
+if (${install_dependencies} == true && ${spack_dependencies} == true); then
```

Comment:
I cut and pasted from other locations. What are the implications of spawning a subshell?

Reply:
The others might work because all they do is evaluate a variable in the subshell, and I guess the return value of that subshell is meaningful (?), but the `==` operator is provided by the bash test built-in. I think you need to be inside `[[ ]]` brackets for it to work as expected.

[Shellcheck](https://www.shellcheck.net/) is great static analysis tool for bash scripts, btw. The warnings are colour-coded by severity and each warning has a code with a wiki page explaining the problem in more detail. A local version of the tool is just an apt-get away, too. This particular issue is [SC2205](https://github.com/koalaman/shellcheck/wiki/SC2205), (though it only describes the sh fix using `[ ]` and we're fine with using the `[[ ]]` bash extensions).

Reply:
Oh. I know what those subshells are doing. The values of variables are `true` and `false`. The substitution is done before the subshell is started, so it's basically doing the equivalent of calling `bash -c 'true'` (which is like calling `/usr/bin/true`), or `bash -c 'false'` (which is like calling `/usr/bin/false`).